wfe/web-front-end.go

```diff
@@ -169,27 +183,29 @@ func (wfe *WebFrontEndImpl) HandleFunc(mux *http.ServeMux, pattern string, h fun
 		if err == nil {
 			response.Header().Set("Replay-Nonce", nonce)
 		}
-		response.Header().Set("Access-Control-Allow-Origin", "*")
 
-		// Return a bodyless response to HEAD for any resource that allows GET.
-		if _, ok := methodsOK["GET"]; ok && request.Method == "HEAD" {
-			// We'll be sending an error anyway, but we
-			// should still comply with HTTP spec by not
+		switch request.Method {
+		case "HEAD":
```

Comment:
Right. Added "bodyless" and "405" checks in the "HEAD doesn't work with POST-only endpoints" test.